src/BalanceTx/Collateral.purs

```diff
@@ -218,6 +224,7 @@ selectCollateral maxCollateralInputs =
     -- with the number of utxos <= `maxCollateralInputs`:
     <<< combinations maxCollateralInputs
     <<< map asTxUnspentOutput
+    <<< List.take maxCandidateUtxos
```

Comment:
I don't think it's acceptable. It's a regression even, because now it's possible that we don't include the only Ada UTxOs and fail because of this limit.

Instead, let's do this:

- use only utxos with ada (and possibly tokens), sorted by amount, in decreasing order 
- in case there's not a single one candidate after we cut 10 first entries in this list, that means we have a lot of ada-only dust, so we can't combine them anyways and we can fail

Reply:
@klntsky It's a good idea to sort utxos by amount in decreasing order before taking the first 10, but there is no need to impose any restrictions on whether an utxo should contain any tokens or not, since we add collateral return output afterwards

Reply:
It will be beneficial in terms of Tx size, right?

Reply:
@klntsky The algorithm will attempt to select the combination with fewer tokens, thus keeping the size of tx relatively small given all the constraints